presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -548,18 +552,30 @@ private static String keyFromPath(Path path)
 
     private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig, ClientConfiguration clientConfig)
     {
+        AmazonS3Client client = null;
         // first try credentials from URI or static properties
         try {
-            return new AmazonS3Client(new StaticCredentialsProvider(getAwsCredentials(uri, hadoopConfig)), clientConfig, METRIC_COLLECTOR);
+            client = new AmazonS3Client(new StaticCredentialsProvider(getAwsCredentials(uri, hadoopConfig)), clientConfig, METRIC_COLLECTOR);
         }
         catch (IllegalArgumentException ignored) {
         }
 
         if (useInstanceCredentials) {
-            return new AmazonS3Client(new InstanceProfileCredentialsProvider(), clientConfig, METRIC_COLLECTOR);
+            client = new AmazonS3Client(new InstanceProfileCredentialsProvider(), clientConfig, METRIC_COLLECTOR);
+        }
+        if (client == null) {
+            throw new RuntimeException("S3 credentials not configured");
+        }
+        final String regionName = hadoopConfig.get(S3_REGION);
+        if (regionName != null) {
+            Region region = Region.getRegion(Regions.fromName(regionName));
+            if (region != null) {
+                client.setRegion(region);
+                client.setEndpoint(region.getServiceEndpoint(ServiceAbbreviations.S3));
```

Comment:
Ah, I see that `Regions` already includes `CN_NORTH_1`, so we don't need to update the library.